### PR TITLE
Note deprecation of constant keyword on functions

### PR DESCRIPTION
**Pull request by wbt**
Per https://github.com/ethereum/solidity/issues/992#issuecomment-322351715

### CODE REVIEW

File: docs/contracts.rst [hunk 1]
```diff
@@ -467,7 +467,7 @@ The following statements are considered modifying the state:
     }
 
 .. note::
-  ``constant`` is an alias to ``view``.
+  ``constant`` is an alias to ``view``, but this is `deprecated <https://github.com/ethereum/solidity/issues/992#issuecomment-322351715>`_ and will be removed in the next breaking release.
```
Comment (chriseth) [inline docs/contracts.rst]: You do not have to cite github issues in the documentation. :)
Please also mention version 0.5.0 explicitly, not only relatively.

**Comment by wbt** [comment]
Thanks for clarifying the version number, I wasn't 100% sure and didn't want to edit in the wrong one.  

Though I appreciate that it's not required, I think the link to the GitHub issue where this was decided is appropriate, as somebody will be following old code/tutorials with `constant`, run into a problem, and want to know when and why the change was made.  The hyperlink is a big step toward answering such questions without any additional screen real estate or reading time.  Also, there are plenty of other links pointing to that issue from other places saying the same thing.

File: docs/contracts.rst [hunk 2]
```diff
@@ -467,7 +467,7 @@ The following statements are considered modifying the state:
     }
 
 .. note::
-  ``constant`` is an alias to ``view``.
+  ``constant`` was an alias to ``view``, but this is `deprecated <https://github.com/ethereum/solidity/issues/992#issuecomment-322351715>`_ and has been removed as of version 0.5.0.
```
Comment (axic) [inline docs/contracts.rst]: There wasn't a 0.5.0 release yet, so this cannot be past tense. Also please do not include the github comment as it is irrelevant. This is the official documentation, no need to prove the content with links to issues.

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
Squashed, rebased and reworded.

**Review by axic — APPROVED** [review]

**Pull request merged**